app/src/ui/preferences/appearance.tsx

```diff
@@ -32,11 +52,54 @@ export class Appearance extends React.Component<IAppearanceProps, {}> {
     }
   }
 
+  private onAutomaticallySwitchThemeChanged = (
+    event: React.FormEvent<HTMLInputElement>
+  ) => {
+    const value = event.currentTarget.checked
+
+    // const updateThemeBasedOnSystem = () => {
```

Comment:
I'm not sure where you plan to encapsulate this logic, but I'd check out [`update-store.ts`](https://github.com/desktop/desktop/blob/master/app/src/ui/lib/update-store.ts) as a reference:

 - exports a singleton (single instance) of a class that wraps the relevant Electron APIs
 - it can subscribe to electron lifecycle events to cleanup after itself
 - other code can `import` this module if they want - they get the same instance